oc/openshift_ops.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from collections import defaultdict
+from kubernetes import config
+from openshift.dynamic import DynamicClient, exceptions
+
+log = logging.getLogger(__name__)
+
+
+class OCP(object):
+    """Class which contains various utility functions for interacting
+    with OpenShift
+
+    """
+    def __init__(self):
+
+        k8s_client = config.new_client_from_config()
+        dyn_client = DynamicClient(k8s_client)
+
+        self.v1_service_list = dyn_client.resources.get(
+            api_version='v1', kind='ServiceList'
+        )
+        self.v1_projects = dyn_client.resources.get(
+            api_version='project.openshift.io/v1', kind='Project'
+        )
+
+    def get_projects(self):
+        """Gets all the projects in the cluster.
+
+        Returns:
+            list: List of projects
+
+        """
+        try:
+            ret = self.v1_projects.get()
+            return [each.metadata.name for each in ret.items]
+        except Exception as err:
+            raise Exception(err)
```

Comment:
Why this try except if you just re-raising the exception without any other handling/logging or do some action?
I think that the behavior will be same if you will omit this try except here.

In case you just reraising the same exception you don't have to store it in err and use just keyword raise which without any parameter will re-raise latest exception.

I think I asked about same syntax in some other PR maybe from someone else but missed the answer. Thanks